### PR TITLE
fix unshare option handling and documentation

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

File: docs/buildah-unshare.md [hunk 1]
```diff
@@ -13,20 +13,41 @@ to be UID 0 and GID 0, respectively.  Any ranges which match that user and
 group in /etc/subuid and /etc/subgid are also mapped in as themselves with the
 help of the *newuidmap(1)* and *newgidmap(1)* helpers.
 
-This is mainly useful for troubleshooting unprivileged operations and for
+buildah unshare is useful for troubleshooting unprivileged operations and for
 manually clearing storage and other data related to images and containers.
 
+It is also useful if you want to use the `buildah mount` command.  If a non
+privileg users wants to mount and work with a container, then they need to execute
```
Comment (nalind) [inline docs/buildah-unshare.md]: "an unprivileged user"

File: docs/buildah-unshare.md [hunk 2]
```diff
@@ -13,20 +13,41 @@ to be UID 0 and GID 0, respectively.  Any ranges which match that user and
...
+buildah unshare.  Executing `buildah mount` is not allowed as a non privileged unless
```
Comment (nalind) [inline docs/buildah-unshare.md]: "as an unprivileged user"

**Comment by rhatdan** [comment]
@giuseppe @vrothberg @TomSweeneyRedHat @nalind @pixdrift PTAL

**Comment by giuseppe** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**